src/adapter/src/catalog/transact.rs

```diff
@@ -458,19 +480,72 @@ impl Catalog {
         let mut storage_collections_to_drop = BTreeSet::new();
 
         for op in ops {
-            Self::transact_op(
+            let (weird_builtin_table_update, temporary_item_actions) = Self::transact_op(
                 oracle_write_ts,
                 session,
                 op,
                 &temporary_ids,
-                builtin_table_updates,
                 audit_events,
                 tx,
                 state,
                 &mut storage_collections_to_create,
                 &mut storage_collections_to_drop,
             )
             .await?;
+
+            // Certain builtin tables are not derived from the durable catalog state, so they need
+            // to be updated ad-hoc based on the current transaction. This is weird and will not
+            // work if we ever want multi-writer catalogs or high availability catalogs. If this
+            // instance crashes after committing the durable catalog but before applying the weird
+            // builtin table updates, then they'll be lost forever.
```

Comment:
This comment should also address the correctness implications of losing those updates. Are things messed up? Is it always log/usage data but never SQL objects?

Reply:
I updated this comment to indicate that there are no correctness issues with the current design and that there would only be correctness issues if we tried to implement a multi-writer catalog or high availability catalog architecture without changing this section. It's hard to say exactly what the correctness implications would be without actually designing that architecture.

How does the new comment look?

Reply:
New comment lgtm.